llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -4163,7 +4293,9 @@ static Value *foldICmpWithLowBitMaskedVal(ICmpInst &I,
     M = Constant::replaceUndefsWith(VecC, SafeReplacementConstant);
   }
 
-  return Builder.CreateICmp(DstPred, X, M);
+  if (NeedsNot)
+    M = IC.Builder.CreateNot(M);
```

Comment:
Use `IC.getFreelyInverted`?

Reply:
If its freely invertable `(not M)` will simplify later on. Seems simpler to let the existing pipeline clean it up. No?